specs/agents/otel-distribution.md

```diff
@@ -0,0 +1,65 @@
+
+## Terminology
+
+**Vanilla OpenTelemetry Distribution**: this is the "upstream" OpenTelemetry distribution that is maintained by the OpenTelemetry community.
+Implementation differs per platform, but it usually consists of an API/SDK and can also provide automatic instrumentation.
+
+**Elastic OpenTelemetry Distribution**: this is an OpenTelemetry distribution provided by Elastic that is derived from the 
+_Vanilla OpenTelemetry Distribution_.
+
+## General guidelines
+
+Those statements are guiding principles of the Elastic OpenTelemetry distributions, they should be considered more as advice than strict rules.
+
+Elastic OpenTelemetry distribution SHOULD ideally:
+- behave as drop-in replacements of their upstream counterparts
+- provide a simple setup and favor onboarding experience (aka "things should work by default").
+- avoid capturing potentially confusing data (see [system metrics](#system-metrics) example below).
+
+## Configuration
+
+Elastic OpenTelemetry distributions MAY override the default configuration.
+When doing so, user-configuration should remain consistent with vanilla distribution:
+- explicit user configuration SHOULD remain effective
+- overriden default configuration MUST have the ability to be restored to upstream default
+
+Elastic specific configuration items MUST be prefixed with `ELASTIC_OTEL_`.
+For example, the [inferred spans](#inferred-spans) feature is configured with `ELASTIC_OTEL_INFERRED_SPANS_*`.
+
+Elastic and platform specific configuration items must be prefixed with `ELASTIC_OTEL_${platform}|` to be consistent with
+the upstream `OTEL_${platform}_` prefix.
+
+For simplicity the configuration in this specification will use the "environment variable" syntax, some platforms like Java
+might also support other ways to configure.
+
+## Features
+
+### Inferred spans
+
+Supported platforms: [Java](https://github.com/elastic/elastic-otel-java/tree/main/inferred-spans)
+
+Configuration namespace: `ELASTIC_OTEL_INFERRED_SPANS_*`
```

Comment:
There are quite a lot of options in the Java agent, do you mean we should move them all here ? Given we only have one implementation for now those might actually be platform-specific and we are not sure all of them will be cross-platform.

Reply:
I think just not leave it at `*`, ie add a specific example

Reply:
For now the inferred spans is java-specific, but the configuration is using the global namespace, so it would make sense to fix this inconsistency by moving everything from `ELASTIC_OTEL_INFERRED_SPANS_` to `ELASTIC_OTEL_JAVA_INFERRED_SPANS_`. However after discussing this with @JonasKunz, since we are planning to also contribute this upstream the options are also likely to change again as part of the process.

So here, instead of fixing this inconsistency it's probably simpler to just omit that in the spec and wait until we have contributed this upstream, so if there are config changes we do it only once.

Reply:
Hopefully the one that enables inferred spans it may be shared :sweat_smile:

Reply:
I've documented this small inconsistency in 6d9c8ca6456b22af8a9df7fe64bbfe3f277328c9, for simplicity I've only added the "enable" config option for now.